s3.js

```diff
@@ -201,3 +206,325 @@ app.get('/upload-presigned-images/:imageId', (req, res) => {
+          res.send("Successfully  updated partition data in ID " + req.body.EventId)
+      }
+  });
+});
+
+// Delete the whole document
+app.post('/delete-event', (req, res)=>{
+  console.log("Deleting event");
+
+  var params = {
+      TableName : "events",
+      Key: {
+        "id": req.body.EventId
+      }
+  };
+
+  console.log("Attempting to delete the whole document...");
+  docClient.delete(params, function(err, data) {
+    if (err) {
+        console.error("Unable to delete the document. Error JSON:", JSON.stringify(err, null, 2));
+        res.send("Failed")
+    } else {
+        console.log("Delete document succeeded:", JSON.stringify(data, null, 2));
+        res.send("Successfully  deleted document with ID " + req.body.EventId)
+    }
+  });
+});
+
+// Deletes item from array in document
+app.post('/delete-partition-data', (req, res)=>{
+  console.log("deleting partition data");
+
+  var params = {
+      TableName : "events",
+      Key: {
+        "id": req.body.EventId
+      },
+      UpdateExpression: "set LayoutData[" + req.body.LayoutDataIndex + "].SceneData.SettingProperties.partitionData[" + req.body.partitionDataIndex + "] = :pd",
+      ExpressionAttributeValues: {
+        ":pd": null
+      },
+      ReturnValues: "NONE"
+  };
+
+  console.log("Attempting to delete partition data...");
+  docClient.update(params, function(err, data) {
+    if (err) {
+        console.error("Unable to delete partition data. Error JSON:", JSON.stringify(err, null, 2));
+        res.send("Failed")
+    } else {
+        console.log("delete partition data succeeded:", JSON.stringify(data, null, 2));
+        res.send("Successfully  deleted an array in document with ID " + req.body.EventId)
+    }
+  });
+});
+
+// Get's the document and then post it with a new event ID
+app.post('/duplicate/:eventId', (req, res) => {
+  console.log("Duplicate event");
+
+
+  var params = {
+      TableName : "events",
+      KeyConditionExpression: "id = :id",
+      ExpressionAttributeValues: {
+        ":id": req.params.eventId
+      }
+  };
+
+  docClient.query(params, function(err, data) {
+      if (err) {
+          console.error("Unable to query. Error:", JSON.stringify(err, null, 2));
+      } else {
+          console.log("Query succeeded.");
+          var document = data.Items[0]
+          document.id = uuidv4()
+          console.log("Adding a new event...");
+          var params = {
+            TableName : "events",
+            Item: document
+          }
+          docClient.put(params, function(err, data) {
+              if (err) {
+                  console.error("Unable to add event. Error JSON:", JSON.stringify(err, null, 2));
+                  res.send("Failed")
+              } else {
+                  console.log("Added item:", JSON.stringify(data, null, 2));
+                  res.send("Successfully added item with ID " + document.id)
```

Comment:
We have 2 x `res.send` here in the event of success - only one should be needed.